src/routes/LPOptimisers/VaultFormRoute/VaultWithdrawRolloverForm/VaultWithdrawRolloverForm.tsx

```diff
@@ -37,12 +41,25 @@ export const VaultWithdrawRolloverForm: React.FunctionComponent<VaultWithdrawRol
   const weights = distribution === 'automatic' ? automaticWeights : manualWeights;
   const combinedWeightValue = weights.reduce((total, weight) => total + weight.distribution, 0);
 
+  const spareWeights = weights
+    .map((w, index): [string, number] => [vault.vaults[index].vaultId, w.distribution])
+    .filter((w) => w[1] > 0);
+
   const withdraw = () => {
-    if (!vault.withdrawable(vaultIndex)) {
+    if (!signer) {
+      return;
+    }
+
+    if (!vault.vaults[vaultIndex].withdrawable) {
       return;
     }
+
     setWithdrawOrRolloverState(WithdrawStates.WITHDRAW_PENDING);
-    void vault.withdraw(vaultIndex).then(
+    void executeWithdraw({
+      optimiserId: vault.optimiserId,
+      vaultId: vault.vaults[vaultIndex].vaultId,
```

Comment:
perhaps vault = vault.vaults[vaultIndex] on top would make for less changes across code

Reply:
96a6f6043cbbb68b1f36d330e8cecdd5ce4c5384